juci/source.cc

```diff
@@ -346,6 +349,52 @@ void Source::ClangView::update_syntax(const std::vector<Source::Range> &ranges)
   }
 }
 
+void Source::ClangView::update_diagnostics() {
+  diagnostic_tooltips.clear();
+  auto diagnostics=tu_->get_diagnostics();
+  auto buffer=get_source_buffer();
+  for(auto& diagnostic: diagnostics) {
+    if(diagnostic.path==file_path) {
+      auto start=buffer->get_iter_at_offset(diagnostic.start_location.offset);
+      auto end=buffer->get_iter_at_offset(diagnostic.end_location.offset);
+      diagnostic_tooltips.add(diagnostic.severity_spelling+": "+diagnostic.spelling, get_source_buffer()->create_mark(start), get_source_buffer()->create_mark(end));
+      auto tag=buffer->create_tag();
+      tag->property_underline()=Pango::Underline::UNDERLINE_ERROR;
+      if(diagnostic.severity<=CXDiagnostic_Warning) {
+        //TODO: get color from config.json
+        tag->set_property("underline-rgba", Gdk::RGBA("orange"));
```

Comment:
The underline-rgba-property came with GTK+ 3 version 3.16 (https://developer.gnome.org/gtk3/stable/GtkTextTag.html#GtkTextTag--underline-rgba). Either one could check if Gtk::TextTag got this property (I never figured out how to do this), or one could check the GTK+ 3 version before trying to set color on the underline. Will try to get the former solution to work.